src/replace.rs

```diff
@@ -75,6 +70,8 @@ impl Data {
             self.original.len()
         );
 
+        let insert_only = from > up_to_and_including;
```

Comment:
i feel like this should check that its== from+1 and the ensure above should check for that, i think ranges where from is 2 or more are still invalid, not sure if its worth testing for that or its one of those "I assume this will actually never happen" things.